plantcv/plantcv/spectral_index/spectral_index.py

```diff
@@ -0,0 +1,863 @@
+# Extract one of the predefined indices from a hyperspectral datacube
+
+import os
+import numpy as np
+from plantcv.plantcv import params
+from plantcv.plantcv import plot_image
+from plantcv.plantcv import print_image
+from plantcv.plantcv import fatal_error
+from plantcv.plantcv import Spectral_data
+from plantcv.plantcv.transform import rescale
+from plantcv.plantcv.hyperspectral import _find_closest
+
+def ndvi(hsi, distance=20):
+    """Normalized Difference Vegetation Index.
+    The theoretical range for NDVI is [-1.0, 1.0]
+
+    inputs:
+    hsi           = hyperspectral image (PlantCV Spectral_data instance)
+    distance = how lenient to be if the required wavelengths are not available
+
+    Returns:
+    index_array    = Index data as a Spectral_data instance
+    :param hsi: __main__.Spectral_data
+    :param distance: int
+    :return index_array: __main__.Spectral_data
+    """
+    params.device += 1
+
+    if (float(hsi.max_wavelength) + distance) >= 800 and (float(hsi.min_wavelength) - distance) <= 670:
+        # Obtain index that best represents NIR and red bands
+        nir_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 800)
+        red_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 670)
+        nir = (hsi.array_data[:, :, [nir_index]])
+        red = (hsi.array_data[:, :, [red_index]])
+        # Naturally ranges from -1 to 1
+        index_array_raw = (nir - red) / (nir + red)
+        return _package_index(hsi=hsi, raw_index=index_array_raw, method="NDVI")
+    else:
+        fatal_error("Available wavelengths are not suitable for calculating NDVI. Try increasing distance.")
+
+def gdvi(hsi, distance=20):
```

Comment:
GDVI is nir - green, not red. The Sripada et al. paper used G = 520-620, and TERRA-REF uses R580. I think we meant to use R580 here? Since the equation is just nir - green, I'm not sure I see why the range is -2.0 to 2.0.

Reply:
Sure, since these are already there in plantcv so I did not check the references. I have made the changes.